fs/fs_zenfs.cc

```diff
@@ -1171,6 +1172,18 @@ std::map<std::string, std::string> ListZenFileSystems() {
   return zenFileSystems;
 }
 
+void ZenFS::GetZoneSnapshot(std::vector<ZoneSnapshot>& zones) {
+  zbd_->GetZonesSnapshot(zones);
+}
+
+void ZenFS::GetZoneFileSnapshot(std::vector<ZoneFileSnapshot>& zone_files) {
+  files_mtx_.lock();
+  for (auto& file_it : files_) {
```

Comment:
Since the original `files_` should not be modified we should use `auto const& file_it : files_` here.

Reply:
Yes, but I found out when I used ZoneFile::GetID() that it is not a const function, and I am not sure if there is a reason for this design. The same problem happens elsewhere, and I think maybe this would be better as a separate PR.

Reply:
Okay makes sense :)